gprofiler/main.py

```diff
@@ -243,12 +245,21 @@ def verify_preconditions():
     return True
 
 
+def log_system_info():
+    uname = platform.uname()
+    logger.info(f"Kernel uname release: {uname.release}")
+    logger.info(f"Kernel uname version: {uname.version}")
+    logger.info(f"Total CPUs: {os.cpu_count()}")
+    logger.info(f"Total RAM: {psutil.virtual_memory().total / (1 << 30):.2f} GB")
+
```

Comment:
Let's add `platform.linux_distribution()`.

Reply:
Good idea. But no such function

Reply:
There's https://pypi.org/project/distro/

Reply:
But that'd require escaping the container :/ Otherwise we'll just get Ubuntu 20.04.

Easiest would be `nsenter -t 1 -m cat /etc/os-release`

Reply:
`/etc/os-release` isn't enough :(

`distro` does it like this:
```
* The os-release file /etc/os-release, if present.
* The output of the lsb_release command, if available.
* The distro release file (/etc/*(-|_)(release|version)), if present.
```
By the way I'm not sure how important the linux distro is if we're running inside a container...? But probably better safe than sorry.

Reply:
1. I'm thinking ahead, for the pyinstaller version
2. It's important also when we're a container (for example - if OS is alpine and there's some Java running on the host, we need to use musl-based async-profiler, which we currently don't employ. Distro name in the logs will explain that)

Reply:
What I meant was that in the host we can just use `distro`... I do agree that it would be better to just cover all cases.

Reply:
I'll go with this:
1. If `is_same_ns(1, "mnt")` - just run `distro`.
2. If not, `fork()` then `setns` and then run `distro`. Then I need to pipe back the information.

Reply:
Done